templates/commands/goldentest/verify.go

```diff
@@ -60,9 +68,127 @@ func (c *VerifyCommand) Run(ctx context.Context, args []string) error {
 		return fmt.Errorf("failed to parse flags: %w", err)
 	}
 
-	if _, err := parseTestCases(ctx, c.flags.Location, c.flags.TestName); err != nil {
-		return fmt.Errorf("failed to parse golden test: %w", err)
+	testCases, err := parseTestCases(ctx, c.flags.Location, c.flags.TestName)
+	if err != nil {
+		return fmt.Errorf("failed to parse golden tests: %w", err)
+	}
+
+	// Create a temporary directory to render golden tests
+	tempDir, err := renderTestCases(testCases, c.flags.Location)
+	defer os.RemoveAll(tempDir)
+	if err != nil {
+		return fmt.Errorf("failed to render test cases: %w", err)
 	}
 
-	return fmt.Errorf("unimplemented")
+	var merr error
+	// Highlight error message color, given diff text might be hundreds lines long.
+	red := color.New(color.FgRed).SprintFunc()
+	green := color.New(color.FgGreen).SprintFunc()
+	resultReport := "\nTest Report:\n"
+
+	for _, tc := range testCases {
+		goldenDataDir := filepath.Join(c.flags.Location, goldenTestDir, tc.TestName, testDataDir)
+		tempDataDir := filepath.Join(tempDir, goldenTestDir, tc.TestName, testDataDir)
+
+		fileSet := make(map[string]struct{})
+		if err := addTestFile(&fileSet, goldenDataDir); err != nil {
+			return err
+		}
+		if err := addTestFile(&fileSet, tempDataDir); err != nil {
+			return err
+		}
+
+		dmp := diffmatchpatch.New()
+
+		var tcErr error
+		for relPath := range fileSet {
```

Comment:
In Go, the iteration order of a map is undefined, which means the order of the file comparisons is undefined. We probably want to do something deterministic here, so the test output is the same every time. Perhaps alphabetical (which would implicitly also be depth-first if we include the directory name as part of the path)?

Reply:
Good idea! Added.

Reply:
Did this get fixed already? I think I'm still seeing the old version where the order is undefined

Reply:
Yes this is an older snapshots. Being fixed in the latest one.